src/supertux/autotile.cpp

```diff
@@ -116,35 +104,28 @@ uint8_t
 Autotile::get_first_mask() const
 {
   if (!m_masks.empty())
-    return m_masks[0]->get_mask();
+    return m_masks[0].get_mask();
   return 0;
 }
 
-std::vector<std::pair<uint32_t, float>>
-Autotile::get_all_tile_ids() const
-{
-  return m_alt_tiles;
-}
-
-bool
-Autotile::is_solid() const
-{
-  return m_solid;
-}
-
-
 // AutotileSet.
 
-std::vector<AutotileSet*>* AutotileSet::m_autotilesets = new std::vector<AutotileSet*>();
+std::vector<std::unique_ptr<AutotileSet>> AutotileSet::m_autotilesets;
 
-AutotileSet::AutotileSet(std::vector<Autotile*> tiles, uint32_t default_tile, std::string name, bool corner) :
-  m_autotiles(std::move(tiles)),
+AutotileSet::AutotileSet(const std::vector<Autotile*>& tiles, uint32_t default_tile, const std::string& name, bool corner) :
```

Comment:
`Autotile` still seems to be used as a raw pointer.

Reply:
This is intentional. If I changed it to `std::unique_ptr` I wouldn't be able to initialize `m_autotiles` class member.

Reply:
What if `AutotileParser::parse_autotile` returns an `Autotile` object and the vector is just `std::vector<Autotile>`?

Reply:
`Autotile` class is not-copyable.

Reply:
Is there a reason for it not to be?

Reply:
It has quite many members so copying object of that class will definitely take longer than only copying pointer. `AutotileMask` is small enough that it can be copied.

Reply:
Okay.